### PR TITLE
Page and config authors (#2024)

### PR DESCRIPTION
**Pull request by sethm**
## Code changes

This PR is meant to address #2024 by introducing `authors` to Page frontmatter and a default `author` to Config. The built-in RSS and Atom templates have been updated to use either the first available author from the Page's frontmatter, or the default `author` from Config, or the fixed string "Unknown" if neither is available.

### CODE REVIEW

**Comment by obi1kenobi** [comment]
Thanks for putting this together! I'm excited to use it as soon as it's merged. It seems like the tests have failed but only due to clippy, and I believe `cargo clippy --fix` will take care of it.

**Comment by sethm** [comment]
This is a tough one... the clippy warnings that cause the build to fail are from code I didn't touch. It looks like Rust 1.64.0 is happy, but Rust 1.67.0 (current stable) is Running `cargo clippy --fix --workspace`  modifies 32 files but still doesn't clear up all the warnings, so I'm pretty wary of doing a clippy fix in this PR. Does anyone else have any opinions on it?

**Comment by obi1kenobi** [comment]
They are newly-added lints encouraging use of the newer inline formatting ability: `"{foo}"` instead of `"{}", foo`. Many projects have been dealing with this kind of issue, and the lint is essentially cosmetic — not a correctness or API misuse lint.

It's safe to ignore, and it's also an easy fix. I'd be happy to open a PR to fix all the new lints, if you'd like.

File: components/templates/src/builtins/atom.xml
Comment (Keats) [inline components/templates/src/builtins/atom.xml]: You don't need `[0]` in the expression I think

File: components/templates/src/builtins/rss.xml
Comment (Keats) [inline components/templates/src/builtins/rss.xml]: same here

**Review by Keats — COMMENTED** [review]
Don't worry about the clippy issue, it's just some annoying change

File: components/templates/src/builtins/rss.xml
Comment (sethm) [reply components/templates/src/builtins/rss.xml]: I think you're right -- fixed in new push.

**Comment by Keats** [comment]
Seems good to me. Do we need authors for the sections as well?

**Comment by obi1kenobi** [comment]
The clippy lints that were raised have been downgraded to allowed-by-default in the new Rust 1.67.1 that was just released. They were annoying many projects like this, and the lang maintainers listened :)

**Comment by Keats** [comment]
I've just seen that, very nice - it annoyed me in multiple repos already.

**Review by Keats — APPROVED** [review]

**Comment by obi1kenobi** [comment]
Awesome, thank you! Do you happen to have a sense of when this might get released? A few people have reached out to tell me that my RSS is failing validation right now, and this PR is part of the fix so I'd love to deploy it on my site as soon as it's released.

**Comment by Keats** [comment]
Not sure yet, I think not too long. I would have liked to fix some Windows bugs before releasing but I might not wait for contributions (no Windows machine here).

**Comment by sethm** [comment]
> Awesome, thank you! Do you happen to have a sense of when this might get released? A few people have reached out to tell me that my RSS is failing validation right now, and this PR is part of the fix so I'd love to deploy it on my site as soon as it's released.

@obi1kenobi In the meantime, you can do what I've done. I copied the atom.xml (or, in your case, rss.xml) file from the Zola source into my site's `templates` directory, and then edited it to include a hard-coded "author". That will cause Zola to use the local copy instead of the built-in template. It's not a complete fix because you can't set authors on a per-page basis, but it will at least fix your RSS for now!

**Pull request merged**
